### PR TITLE
feat: add top level download command

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

**Review by nugaon — CHANGES_REQUESTED** [review]
Please rewrite the `address` argument's description from "Root manifest reference" to "Manifest reference (with optional path)" or smth.
Also create test for that use-case, when the `address` has this path.

File: src/command/download.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { Aggregation, LeafCommand } from 'furious-commander'
+import { MantarayNode } from 'mantaray-js'
+import { referenceToHex } from '../utils'
+import { BzzAddress } from '../utils/bzz-address'
+import { Download as ManifestDownload } from './manifest/download'
+import { RootCommand } from './root-command'
+
+export class Download extends RootCommand implements LeafCommand {
+  public readonly name = 'download'
+  public readonly aliases = ['dl']
```
[the remark this answers is not in the document]
Comment (Cafe137) [reply src/command/download.ts:10]: I think `dl` is more widespread.

We can include both once I am done with multiple alias support in the core. 🙂

**Review by nugaon — APPROVED** [review]
Nice work!

**Comment by agazso** [comment]
I'm trying to download the Swarm website (437b4a5f61b44ce6f6c69b4841fdec548d2dba1b664b2fb8b02489d680df9207) which has quite big files. The filenames are printed only after the file is downloaded and there is no indication of anything is happening. Also with the `-v` option I only see the `Downloading <filename>` message _after_ the file was downloaded.

It would be good to have at least a spinner or some indicator when the file is being downloaded. With the `-v` option it would be good if the `Downloading...` message were printed _before_ the file is downloaded.

**Pull request merged**